driver-mapping/src/test/java/com/datastax/driver/mapping/MapperAsyncResultTest.java

```diff
@@ -0,0 +1,140 @@
+/*
+ *      Copyright (C) 2012-2015 DataStax Inc.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+package com.datastax.driver.mapping;
+
+import com.datastax.driver.core.CCMTestsSupport;
+import com.datastax.driver.core.SimpleStatement;
+import com.datastax.driver.core.Statement;
+import com.datastax.driver.core.utils.CassandraVersion;
+import com.datastax.driver.mapping.annotations.PartitionKey;
+import com.datastax.driver.mapping.annotations.Table;
+import com.google.common.base.Objects;
+import com.google.common.util.concurrent.AsyncFunction;
+import com.google.common.util.concurrent.Futures;
+import com.google.common.util.concurrent.ListenableFuture;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import java.util.Set;
+import java.util.concurrent.ConcurrentSkipListSet;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * Tests that a Result can be paged asynchronously.
+ *
+ * @jira_ticket JAVA-1157
+ */
+@SuppressWarnings("unused")
+@CassandraVersion(major = 2.0, description = "uses paging")
+public class MapperAsyncResultTest extends CCMTestsSupport {
+
+    @Override
+    public void onTestContextInitialized() {
+        execute("CREATE TABLE users (id int PRIMARY KEY, name text)");
+    }
+
+    @Table(name = "users")
+    static class User {
+
+        @PartitionKey
+        private int id;
+
+        private String name;
+
+        public int getId() {
+            return id;
+        }
+
+        public void setId(int id) {
+            this.id = id;
+        }
+
+        public String getName() {
+            return name;
+        }
+
+        public void setName(String name) {
+            this.name = name;
+        }
+
+        @Override
+        public boolean equals(Object other) {
```

Comment:
Minor remark: strictly speaking you don't need to implement `equals` and `hashCode` here, since you're accumulating the ids below.